src/views/components/injected/CourseCatalogInjectedPopup/Description.tsx

```diff
@@ -1,37 +1,87 @@
 import Text from '@views/components/common/Text/Text';
 import clsx from 'clsx';
 import React from 'react';
+import type { Course } from 'src/shared/types/Course';
+import { CourseCatalogScraper } from 'src/views/lib/CourseCatalogScraper';
+import { SiteSupport } from 'src/views/lib/getSiteSupport';
+
+import Spinner from '../../common/Spinner/Spinner';
 
 interface DescriptionProps {
-    lines: string[];
+    course: Course;
+}
+
+const LoadStatus = {
+    LOADING: 'LOADING',
+    DONE: 'DONE',
+    ERROR: 'ERROR',
+} as const;
+type LoadStatusType = keyof typeof LoadStatus;
```

Comment:
Ehh almost. This should create a union type of all values of the const object.

```suggestion
type LoadStatusType = (typeof LoadStatus)[keyof typeof LoadStatus];
```